packages/jest-cli/src/TestRunner.js

```diff
@@ -238,10 +247,22 @@ class TestRunner {
         aggregatedResults.success =
           aggregatedResults.numFailedTests === 0 &&
           aggregatedResults.numRuntimeErrorTestSuites === 0;
-        if (reporter.onRunComplete) {
-          reporter.onRunComplete(config, aggregatedResults);
-        }
-        return aggregatedResults;
+
+        return this._hasteMap.matchFiles(/\.snap/).then(res => {
+          aggregatedResults.snapshotsUnlinked = 0;
+          res.forEach(snapFile => {
+            const parts = snapFile.split('__snapshots__/');
+            const testFile = parts[0] + parts[1].slice(0, -5);
+            if (!fileExists(testFile)) {
+              fs.unlinkSync(snapFile);
+              aggregatedResults.snapshotsUnlinked++;
```

Comment:
I'd prefer not to change the state of the file system if the update flag isn't given (with the exception of adding new snapshots). I was thinking of adding another result to aggregatedResults, yes. The snapshot summary can then print an explanation that says something like "One test file removed but snapshot file still exists. Run with jest -u to remove the snapshot file."